docs/2_skeleton.md

```diff
@@ -0,0 +1,169 @@
+# 🧩 Skeleton of a [ReVanced patch](https://github.com/revanced/revanced-patches)
+
+Patches are what make ReVanced, ReVanced. On the following page the basic structure of a patch will be explained.
+
+## ⛳️ Example patch
+
+This page works with the following patch as an example:
+
+```kt
+package app.revanced.patches.ads.patch
+
+// Imports
+
+@Patch
+@Name("disable-ads")
+@Description("Disables ads.")
+@DependsOn([HideAdResourcePatch:class])
+@Compatibility([Package("com.some.app", arrayOf("0.1.0"))])
+@Version("0.0.1")
+class DisableAdsPatch : BytecodePatch(
+    listOf(LoadAdsFingerprint)
+) {
+    override fun execute(context: BytecodeContext): PatchResult {
+        val result = LoadAdsFingerprint.result
+            ?: return PatchResultError("LoadAdsFingerprint not found")
+
+        result.mutableMethod.replaceInstructions(
+                0,
+                """
+                    const/4 v0, 0x1
+                    return v0
+                """
+        )
+
+        return PatchResultSuccess()
+    }
+}
+```
+
+## 🔎 Dissecting the example patch
+
+Lets start with understanding, how a patch is structured. A patch is mainly built out of three components:
+
+1. 📝 Patch annotations
+
+   ```kt
+   @Patch
+   @Name("disable-ads")
+   @Description("Disables ads.")
+   @DependsOn([HideAdResourcePatch:class])
+   @Compatibility([Package("com.some.app", arrayOf("0.1.0"))])
+   @Version("0.0.1")
+   ```
+
+   To give context about the patch, annotations are used. They serve different but important purposes:
+
+   - Every visible patch **should** be annotated with `@Patch` to be picked up by `PatchBundle` from the [introduction](1_introduction.md). Patches which are not annotated with `@Patch` can be referenced by other patches. We refer to those as _patch dependencies_. Patch dependencies are useful to structure multiple patches.
+
+     Example: _To add settings switches to an app, first, a patch is required which can provide a basic framework for other patches to add their toggles to that app. For that, those patches refer to the dependency patch and use its framework to add their toggles to an app. The [ReVanced patcher](https://github.com/revanced/revanced-patcher) will then first execute the dependency and then the patch itself. The dependency can prepare a preference screen when executed and then initialize the its framework for further use by other patches._
+
+   - Visible patches **should** be annotated with `@Name`. This annotation does not serve any functional purpose. Instead, it allows to refer to the patch with a name. The [ReVanced patches](https://github.com/revanced/revanced-patches) use _Kebab casing_ by convention, but any name can be used for patches. Patches with no `@Patch` annotation do not require the `@Name` annotation, because they are only useable as dependencies for other patches, and therefore are not visible through `PatchBundle`.
```

Comment:
```suggestion
   - Visible patches **should** be annotated with `@Name`. This annotation does not serve any functional purpose. Instead, it allows referring to the patch with a name. The [revanced-patches](https://github.com/revanced/revanced-patches) repository uses _kebab-casing_ by convention, but any name can be used for patches. Patches with no `@Patch` annotation do not require the `@Name` annotation, because they are only useable as dependencies for other patches, and therefore are not visible through `PatchBundle`.
```